demmt/object_gf100_compute.c

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright (C) 2015 Samuel Pitoiset <samuel.pitoiset@gmail.com>
+ * All Rights Reserved.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a
+ * copy of this software and associated documentation files (the "Software"),
+ * to deal in the Software without restriction, including without limitation
+ * the rights to use, copy, modify, merge, publish, distribute, sublicense,
+ * and/or sell copies of the Software, and to permit persons to whom the
+ * Software is furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice (including the next
+ * paragraph) shall be included in all copies or substantial portions of the
+ * Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL
+ * THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+ * OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+ * ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ * OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+#include "buffer.h"
+#include "config.h"
+#include "demmt.h"
+#include "log.h"
+#include "nvrm.h"
+#include "object.h"
+
+struct gf100_compute_data
+{
+	struct addr_n_buf code;
+	struct addr_n_buf cb;
+	int data_offset;
+
+	struct mthd2addr *addresses;
+};
+
+static void destroy_gf100_compute_data(struct gpu_object *obj)
+{
+	struct gf100_compute_data *d = obj->class_data;
+	free(d->addresses);
+	free(d);
+}
+
+void decode_gf100_compute_init(struct gpu_object *obj)
+{
+	struct gf100_compute_data *d = obj->class_data = calloc(1, sizeof(struct gf100_compute_data));
+	obj->class_data_destroy = destroy_gf100_compute_data;
+
+#define SZ 3
```

Comment:
Please hook up the texture and image stuff. At least track the image addresses, and print the TSC/TIC's same as gf100_3d does.